presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/UnwrapDateFunctionInPredicate.java

```diff
@@ -0,0 +1,249 @@
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.common.type.DateType;
+import com.facebook.presto.common.type.TimestampType;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.spi.function.StandardFunctionResolution;
+import com.facebook.presto.spi.plan.FilterNode;
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.InputReferenceExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.relational.FunctionResolution;
+import com.facebook.presto.common.function.OperatorType;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+
+import java.util.Optional;
+
+import static com.facebook.presto.matching.Pattern.typeOf;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.AND;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This rule rewrites predicates of the form:
+ *    date(timestamp_column) = DATE 'YYYY-MM-DD'
+ * into:
+ *    timestamp_column >= TIMESTAMP 'YYYY-MM-DD 00:00:00.000'
+ * AND timestamp_column < TIMESTAMP 'YYYY-MM-DD+1 00:00:00.000'
+ */
+public class UnwrapDateFunctionInPredicate
+        implements Rule<FilterNode>
+{
+    private static final String DATE_FUNCTION = "date";
+
+    private final FunctionAndTypeManager functionAndTypeManager;
+    private final StandardFunctionResolution functionResolution;
+
+    public UnwrapDateFunctionInPredicate(FunctionAndTypeManager functionAndTypeManager)
+    {
+        this.functionAndTypeManager = requireNonNull(functionAndTypeManager, "functionAndTypeManager is null");
+        this.functionResolution = new FunctionResolution(functionAndTypeManager.getFunctionAndTypeResolver());
+    }
+
+    @Override
+    public Pattern<FilterNode> getPattern()
+    {
+        return typeOf(FilterNode.class);
+    }
+
+    @Override
+    public Result apply(FilterNode node, Captures captures, Context context)
+    {
+        RowExpression predicate = node.getPredicate();
+        System.out.println("Original Predicate: " + predicate);
```

Comment:
This is fine to leave in for now, but in general we try not to commit any code that was used for debugging locally. It will need to be removed when we make the PR to the main repository